vobject/icalendar.py

```diff
@@ -330,6 +330,8 @@ def pickTzid(tzinfo, allowUTC=False):
         """
         Given a tzinfo class, use known APIs to determine TZID, or use tzname.
         """
+        if str(tzinfo) == 'UTC':
+            return None
```

Comment:
I suggest merging this clause into the next which has a comment about UTC.

Reply:
My patch was:
```
--- a/vobject/icalendar.py
+++ b/vobject/icalendar.py
@@ -332,7 +332,7 @@ class TimezoneComponent(Component):
         """

         # If tzinfo is UTC, we don't need a TZID
-        if tzinfo is None or (not allowUTC and tzinfo_eq(tzinfo, utc)):
+        if tzinfo is None or str(tzinfo) == 'UTC' or (not allowUTC and tzinfo_eq(tzinfo, utc)):
             return None

         # Try pytz tzid key
```

Reply:
This is strange! Just to exclude a weird case: Have you tried
```
if str(tzinfo) == 'UTC' of tzinfo is None or (not allowUTC and tzinfo_eq(tzinfo, utc)):
```